mllib/src/main/scala/org/apache/spark/mllib/recommendation/MatrixFactorizationModel.scala

```diff
@@ -75,10 +75,22 @@ class MatrixFactorizationModel @Since("0.8.0") (
     }
   }
 
+  /** Check for the invalid user. */
+  private def validateUser(user: Int): Unit = {
+    require(userFeatures.lookup(user).nonEmpty, s"userId: $user not found in the model")
```

Comment:
The problem is you now look up the user and product twice. Instead, the callers should probably check whether `userFeatures.lookup(user)` is empty before calling `.head` and otherwise throw the exception.

Reply:
I have renamed the method to 'validateAndGetUser', where it check, whether the user exist or not and it returns the corresponding user feature. Similarly for the product also.
Please let me know if anymore changes required.